src/cat-data-service/src/service/poem_service.rs

```diff
@@ -2,33 +2,26 @@
 //!
 //! This provides only the primary entrypoint to the service.
 
-use crate::service::docs::{docs, favicon};
-use crate::service::Error;
-
 use crate::service::api::mk_api;
+use crate::service::docs::{docs, favicon};
 use crate::service::utilities::catch_panic::{set_panic_hook, ServicePanicHandler};
 use crate::service::utilities::middleware::{
     chain_axum::ChainAxum,
     tracing_mw::{init_prometheus, Tracing},
 };
-
+use crate::service::Error;
 use crate::settings::{get_api_hostnames, API_URL_PREFIX};
 use crate::state::State;
-
 use poem::endpoint::PrometheusExporter;
 use poem::listener::TcpListener;
 use poem::middleware::{CatchPanic, Compression, Cors};
 use poem::web::CompressionLevel;
-use poem::{EndpointExt, IntoEndpoint, Route};
+use poem::{Endpoint, EndpointExt, Route};
 use std::net::SocketAddr;
 use std::sync::Arc;
 
 /// This exists to allow us to add extra routes to the service for testing purposes.
-pub(crate) fn mk_app(
-    hosts: Vec<String>,
-    base_route: Option<Route>,
-    state: Arc<State>,
-) -> impl IntoEndpoint {
+pub fn mk_app(hosts: Vec<String>, base_route: Option<Route>, state: Arc<State>) -> impl Endpoint {
```

Comment:
I have removed `pub(crate)` here, because it is easier to maintain visibility on the module level, which I think much more suitable in this case.

Reply:
Not sure I agree.  `pub` is implicitly private to the crate, depending on the visibility of a parent module which is not unambiguous at the time visibility is set.  Its also fragile, if someone ina PR makes the parent module Public for any reason all of a sudden things which should NOT be public become public.

In code review if I see `pub` I have no way to know is this actually public, or just public to the internals of the crate. 

IMHO its a bad part of the rust spec to rely on parent visibility to control what `pub` may or may not mean.

For example, It's not clear in a review if the intention here is to make the function part of an external API, or private but accessible within the crate itself.  And this should never be part of the public API.  In fact currently, `cat-data-service` should have no public API because its not a library and isn't designed to be consumed as one.

Lets discuss at the Friday or Monday tech meeting, because I think we haven't done a very good job to-date of controlling visibility outside our individual crates.